classes/bridges/across.js

```diff
@@ -19,16 +24,52 @@ class AcrossBridge extends BaseBridge {
     "op mainnet": "0x6f26Bf09B1C792e3228e5467807a900A503c0281",
     polygon: "0x9295ee1d8C5b022Be115A2AD3c30C72E34e7F096",
   };
+
   constructor() {
     super("across");
     this.sdk = createAcrossClient({
       integratorId: "0xaaff", // 2-byte hex string
       chains: [optimism, arbitrum, polygon, base],
     });
+    this.isInitialized = false;
+    this.feeCosts = null;
   }
+
   async init() {
     return;
   }
+
+  async fetchFeeCosts(
+    account,
+    fromChainId,
+    toChainId,
+    inputToken,
+    targetToken,
+    inputAmount,
+    tokenPrices,
+  ) {
+    const route = {
+      originChainId: fromChainId,
+      destinationChainId: toChainId,
+      inputToken: inputToken,
+      outputToken: targetToken,
+    };
+    const quote = await this.sdk.getQuote({
+      route,
+      inputAmount: inputAmount,
+    });
+    const tokenDecimal = await getTokenDecimal(
+      quote.deposit.inputToken,
+      CHAIN_ID_TO_CHAIN[fromChainId].name,
+    );
```

Comment:
minor, but actually we already know all the token's decimals. Fetching the decimal again would slow down the process but it's minor. We can refactor it during cool-down